Update to app layout as per design

nisrulz opened the pull request:
Have a lot of changes, so I believe someone would have to simply download my branch for code review. Summary of changes:
- Replaced navigation drawer with proper implementation
- Setup a Splash Activity
- Moved twitter keys to `secrets.xml` file. It will not be added to gitignore, so if you need it I can share it on request
- Created MyApplication class to init Fabric and Twitter
- Setup COC, About Us, Social screens i.e fragments
- Created blank stubs for MySchedule, Settings, Speakers screens i.e fragments
- COC screen loads a local `html` file for now, but next iteration will have it load it from a url if there is internet otherwise load it from local
- Cleaned up a lot of unused layouts and icons. Added new ones
- Fixes style for the app theme and the tab layout
- fixed style of agenda screen in regards to padding and margins and also agenda detail screen

emezias commented:
@nisrulz Why is there a Splash Activity? 
https://material.io/guidelines/patterns/launch-screens.html#launch-screens-branded-launch
There is almost zero load time because we have the agenda data inside the apk. Let's put the user straight into something useful, Like the speaker list or the agenda view pager.

emezias commented:
The navigation drawer says "Hello World"
The branding image above that text is not working.  A simple title that says "Droidcon Boston '18" or less would be a much better experience. 

After running and trying it I will repeat my thinking about the splash. The activity transition animation is a waste of resources in a resource-constrained mobile device environment - creating an activity just to throw it away? Please don't create unused objects to immediately throw into garbage collection and waste my CPU cycles and battery time. Less is more.

Having said that, the splash might be useful to show when bringing up the AR feature or loading chat (assuming there is enough delay with either). @kimj - do you think it will be useful there? My bet is that the local Firebase data will come up instantly and load and refresh the view in real time.

emezias commented:
The proposed detail page design has issues. Here's a screenshot in case you didn't run the branch. My "large" font size setting may contribute to the problems reported here.

![eric1](https://user-images.githubusercontent.com/830129/36083793-d6bd43fa-0f84-11e8-92e4-aa6a7b87c579.png)

1. The grayscale image is cutting speaker's heads off. 
2. There's no reason to put the same photo on the page twice. 
3. The title of the talk, the speaker name and the speaker's job title are difficult to read.
4. The wrap on the talk title with a right gravity is unexpected 
5. The green on gray icons do not have enough contrast, they need to be sharp and clear. On the desktop screenshot they look fine - on the device the icons are light green blobs.

This screenshot is the final design we settled on for 2017 with the uncut speaker photo and their social links featured at the top as high importance items. The text is clear and easy to read. There's only a small bit of accent color on the page. The iconography is easily understood.

![eric2](https://user-images.githubusercontent.com/830129/36083794-d6d2638e-0f84-11e8-8d97-947442aef3f1.png)

nisrulz commented:
@emezias I am going to address all your comments one by one.
Please note, the code pushed before my last 2 commits is an alpha version and was trying to push something that was close to the design which was submitted earlier as quickly as possible. Apologies if it did not conform to what you were expecting.

Ok, now for your comments.
- Splash Screen: I agree with your point about having an Activity is too much, but I differ from the point where it shouldn't be there. 
One it was already mentioned in the designs submitted earlier and everyone agreed to it, at least no one objected at that time. So in my recent commit, I have removed the Splash activity and put in an Imageview with the splash image which goes away after 2 seconds. 
Second, the reason we have Splash screen is for branding purpose. The link you shared mentions it too from the material design specs. Though we can remove it altogether, as I said I was trying to match the designs submitted earlier.

-  Navigation Drawer: Replaced the text with "Droidcon Boston'18" and the background image of nav header
 <img height="560" src="https://user-images.githubusercontent.com/2096087/36097721-22cebe7a-1022-11e8-9654-d26ac617d50e.png" /> 

-  Agenda Detail's Screen: Again was trying to follow the design submitted earlier. Instead of writing down what has changed let me share a screenshot of the current changes.
 <img height="560" src="https://user-images.githubusercontent.com/2096087/36097524-55bc4920-1021-11e8-88de-7582d375ffbf.png" /> 

## Other screenshots for anyone who hasn't run the app for this branch
<img height="320" src="https://user-images.githubusercontent.com/2096087/36097567-7555007e-1021-11e8-8627-ff18630917a7.png" /> <img height="320" src="https://user-images.githubusercontent.com/2096087/36097568-75980be4-1021-11e8-88c1-04ca6e0e6886.png" /> <img height="320" src="https://user-images.githubusercontent.com/2096087/36097573-7b5aa618-1021-11e8-863f-d52dfe9bfb62.png" />  <img height="320" src="https://user-images.githubusercontent.com/2096087/36097574-7ba28c94-1021-11e8-9fc3-63f1b5267ebc.png" />   <img height="320" src="https://user-images.githubusercontent.com/2096087/36097575-7be4aaf2-1021-11e8-880b-b5dd1b6103d2.png" />

emezias commented:
Sorry Nishant but I missed any designs submitted so you got no feedback
from me until the branch got pushed.

Splash screens are a design artifact of older device hardware, especially
when there is 3G network latency. I am opposed to introducing delay. Other
features of the app might need it. I prefer it does not happen at app
launch.

Some screen backgrounds are gray. Some are white. Can you please make that
consistent?

The social icon links on the speaker detail page are primary content that
can be pushed up to the top so they are visible (and tappable) without
scrolling. This was an important design decision agreed upon with last
year's design. Please make that change.

The larger photo will be better than the circular cropped image from the
page. The current large black margin on top does nothing but introduce
scroll.

I apologize for missing the opportunity to, it's on your design document.

On Feb 12, 2018 7:56 AM, "Nishant Srivastava" <notifications@github.com>
wrote:

@emezias <https://github.com/emezias> I am going to address all your
comments one by one.
Please note, the code pushed before my last 2 commits is an alpha version
and was trying to push something that was close to the design which was
submitted earlier as quickly as possible. Apologies if it did not conform
to what you were expecting.

Ok, now for your comments.

   -

   Splash Screen: I agree with your point about having an Activity is too
   much, but I differ from the point where it shouldn't be there.
   One it was already mentioned in the designs submitted earlier and
   everyone agreed to it, at least no one objected at that time. So in my
   recent commit, I have removed the Splash activity and put in an Imageview
   with the splash image which goes away after 2 seconds.
   Second, the reason we have Splash screen is for branding purpose. The
   link you shared mentions it too from the material design specs. Though we
   can remove it altogether, as I said I was trying to match the designs
   submitted earlier.
   -

   Navigation Drawer: Replaced the text with "Droidcon Boston'18" and the
   background image of nav header

<https://user-images.githubusercontent.com/2096087/36097721-22cebe7a-1022-11e8-9654-d26ac617d50e.png>

   - Agenda Detail's Screen: Again was trying to follow the design
   submitted earlier. Instead of writing down what has changed let me share a
   screenshot of the current changes.

<https://user-images.githubusercontent.com/2096087/36097524-55bc4920-1021-11e8-88de-7582d375ffbf.png>
Other screenshots for anyone who hasn't run the app for this branch

<https://user-images.githubusercontent.com/2096087/36097567-7555007e-1021-11e8-8627-ff18630917a7.png>
<https://user-images.githubusercontent.com/2096087/36097568-75980be4-1021-11e8-88c1-04ca6e0e6886.png>
<https://user-images.githubusercontent.com/2096087/36097573-7b5aa618-1021-11e8-863f-d52dfe9bfb62.png>
<https://user-images.githubusercontent.com/2096087/36097574-7ba28c94-1021-11e8-9fc3-63f1b5267ebc.png>
<https://user-images.githubusercontent.com/2096087/36097575-7be4aaf2-1021-11e8-880b-b5dd1b6103d2.png>

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/Droidcon-Boston/conference-app-android/pull/48#issuecomment-364914868>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAyqsY6mDs8MbCUOj0qyt4mjEDOcDvdEks5tUDTqgaJpZM4SBPgj>
.

emezias commented:
@nisrulz It looks like you want a collapsing toolbar layout with parallax animation from the larger speaker image to the smaller cropped image. I can help you with that code. It's tricky. Get the top margin right so that the toolbar doesn't cut off the speakers heads. Use the full-size headshot and leave out the collapsing part for now. It's important that the app show the headshot without cropping somewhere!

You can push forward on the other pages and changes - especially the change to show the social links at the top of the scrolling content. You can open a GitHub issue for collapsing toolbar and we can do a separate PR for the proper scrolling after merging this one.

Please also change the empty splash at app launch -> go straight to a feature. Branding is helpful for businesses. A lot of apps in the store are intended to promote a brand and have no monetization schemes. We are volunteers. Branding does not add value in our case. This app is going to be used for a couple of weeks. The brand and design will change again for 2019.

nisrulz commented:
@emezias Got all your points. Made a few changes.
 2 things, the one show is the agenda detail. I was going to have a separate speaker detail. I think you are mixing the two. 
Quick question: Do you want me to add the social links to the agenda detail too, because they will be there in the speaker detail with complete detail about the speaker. In Agenda Detail, there is no speaker detail if you didnot notice. There is only the small picture, name and designation. No full description, which was going to be there in the Speaker Detail screen, which people will reach through either clicking the name right in the Agenda Detail or via the Speaker List screen.

Please share your thoughts on the same and clarify.

Thank you

kimj commented:
might make the app simpler to have the talk detail and the speaker detail
on the same page unless it looks like itll be too much text.

On Mon, Feb 12, 2018 at 3:46 PM, Nishant Srivastava <
notifications@github.com> wrote:

> @emezias <https://github.com/emezias> Got all your points. Made a few
> changes.
> 2 things, the one show is the agenda detail. I was going to have a
> separate speaker detail. I think you are mixing the two.
> Quick question: Do you want me to add the social links to the agenda
> detail too, because they will be there in the speaker detail with complete
> detail about the speaker. In Agenda Detail, there is no speaker detail if
> you didnot notice. There is only the small picture, name and designation.
> No full description, which was going to be there in the Speaker Detail
> screen, which people will reach through either clicking the name right in
> the Agenda Detail or via the Speaker List screen.
>
> Please share your thoughts on the same and clarify.
>
> Thank you
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Droidcon-Boston/conference-app-android/pull/48#issuecomment-365056341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAHBoIc2wpddCsKU-L0u9ixmSFuEAAMks5tUKMigaJpZM4SBPgj>
> .
>



-- 
\m/ >-< \m/

nisrulz commented:
@kimj  I think it will be too much text. Also if say we shifted the Speaker description to the Agenda Detail, I believe we dont need the speaker list anymore. What do you suggest? 

Sidenote:[Here is the link to the design mocks](https://drive.google.com/drive/folders/1PDx961StJpzCukdJi1TrgbGA1gDFvYGQ).

emezias commented:
Hey - the agenda detail page still has issues. 
The entire page should scroll. 
The black title area is too large. 
The page needs to show the full headshot, not the cropped icon version from the list.
The social icons for the speaker need to be on this page at the top, visible without a scroll

Also noticed that something broke with the 9am Day 1 session. The display in the agenda is now empty.

The settings page is a little ahead of the app. That should go into a separate request until these other features are available. Don't think that "dark" switch is going to work at all.

kimj commented on Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/utils/DividerItemDecoration.java:
```diff
@@ -0,0 +1,92 @@
+      mDivider.draw(c);
+    }
+  }
+
```
you can merge these methods together and use a method parameter to switch between the two.

emezias commented:
As for the Social page - please put those links directly into the navigation drawer instead of creating an empty page with the sole purpose of opening another page

kimj commented on Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/MainActivity.java:
```diff
@@ -4,160 +4,157 @@
-    final FragmentManager fragmentManager = getSupportFragmentManager();
+  @BindView(R.id.navView)
+  NavigationView navigationView;
 
```
annotation and the field can go on the same line

kimj commented on Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/TweetsFragment.java:
```diff
@@ -20,53 +19,55 @@
+      adapter.refresh(new Callback<TimelineResult<Tweet>>() {
+        @Override
+        public void failure(TwitterException exception) {
+          Toast.makeText(context, "No Tweets", Toast.LENGTH_SHORT).show();
```
change this to error.
